### PR TITLE
Add automated terratest to circleci build

### PR DESCRIPTION
**Pull request by jsclarridge**
Update the CircleCI config to run automated terratests for this module.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,8 +20,21 @@ jobs:
           paths:
             - ~/.cache
 
+  terratest:
+    docker:
+      - image: *circleci_docker_primary
+        environment:
+          AWS_DEFAULT_REGION: us-west-2
```
Comment (dynamike) [inline .circleci/config.yml]: Is this needed?

Comment (jsclarridge) [reply .circleci/config.yml]: No, it's been removed.

Comment (jsclarridge) [reply .circleci/config.yml]: The random region assignment is also working without TERRATEST_REGION. Should we leave TERRATEST_REGION out?

Comment (dynamike) [reply .circleci/config.yml]: I'm okay with leaving it out for now.

**Pull request merged**